Java/core/src/main/java/com/amazon/randomcutforest/executor/AbstractForestUpdateExecutor.java

```diff
@@ -0,0 +1,233 @@
+     *         update.
+     */
+    protected abstract List<Optional<UpdateReturn<P>>> update(P updateInput, long currentIndex);
+
+    /**
+     * Returns a clean deep copy of the point.
+     *
+     * Current clean-ups include changing negative zero -0.0 to positive zero 0.0.
+     *
+     * @param point The original data point.
+     * @return a clean deep copy of the original point.
+     */
+    protected double[] cleanCopy(double[] point) {
+        double[] pointCopy = Arrays.copyOf(point, point.length);
+        for (int i = 0; i < point.length; i++) {
+            if (pointCopy[i] == 0.0) {
+                pointCopy[i] = 0.0;
+            }
+        }
+        return pointCopy;
+    }
+
+    /**
+     *
+     * @return the weighted samples (without sequential information)
+     */
+    public ArrayList<List<Weighted<P>>> getWeightedSamples() {
+        ArrayList<List<Weighted<P>>> result = new ArrayList<>();
+        for (IUpdatable<P> t : models) {
+            result.add(t.getWeightedSamples());
+        }
+        ;
+        return result;
+    }
+
+    /**
+     *
+     * @return the sequential samples; if that is not stored then a dummy value is
+     *         placed
+     */
+    public ArrayList<List<Sequential<P>>> getSequentialSamples() {
+        ArrayList<List<Sequential<P>>> result = new ArrayList<>();
+        for (IUpdatable<P> t : models) {
+            result.add(t.getSequentialSamples());
+        }
+        ;
+        return result;
+    }
+
+    /**
+     * Gets the tree data for compact RCF
+     * 
+     * @return the tree data for the specifica sampler combination
+     */
+    public ArrayList<TreeData> getTreeData() {
+        ArrayList<TreeData> result = new ArrayList<>();
+        for (IUpdatable<P> t : models) {
+            result.add(((SamplerPlusTree) t).getTreeData());
+        }
+        return result;
+    }
+
+    /**
+     * gets the sampler data for compact RCF
+     * 
+     * @return
+     */
+    public ArrayList<CompactSamplerData> getCompactSamplerData() {
+        ArrayList<CompactSamplerData> result = new ArrayList<>();
+        for (IUpdatable<P> t : models) {
+            result.add(((SamplerPlusTree) t).getCompactSamplerData());
+        }
+        return result;
+    }
+
+    /**
+     * gets the pointstore information, currently for Integer refs and double[]
+     * input
+     * 
+     * @return pointstore information,
+     */
+    public PointStoreDoubleData getPointStoredata() {
+        return updateCoordinator.getPointStoreData();
+    }
+
+    /**
+     * initializes the models (sampler + plus) based on samples
+     * 
+     * @param samplerData           data without sequence information
+     * @param sequentialSamplerData data with sequence information Exactly one of
+     *                              the arguments is required.
+     */
+
+    public void initializeModels(ArrayList<List<Weighted<P>>> samplerData,
```

Comment:
Open to it - I think with a more careful refactoring this would help. The issue is "do we want the user to have the flexibility to just use the sampler info and not the trees". I feel we should; the trees can be rebuilt and if that is acceptable to the user then the serialization is smaller in space.